src/core/mavsdk_impl.cpp

```diff
@@ -125,13 +125,6 @@ void MavsdkImpl::receive_message(mavlink_message_t& message)
         std::swap(_systems[message.sysid], it->second);
         _systems[message.sysid]->system_impl()->set_system_id(message.sysid);
         _systems.erase(it);
-    } else if (_is_single_system) {
-        auto it_begin = _systems.begin();
-        if (it_begin->first != message.sysid) {
-            std::swap(_systems[message.sysid], it_begin->second);
-            _systems[message.sysid]->system_impl()->set_system_id(message.sysid);
-            _systems.erase(it_begin);
-        }
```

Comment:
Hm, how come this is not required anymore?

Reply:
Maybe I did not understand well the purpose of this part, can you explain what it does ?

Reply:
Ok I see, this what is done for the first condition, when it founds a system 0:
```
if (it != _systems.end()) {
    std::swap(_systems[message.sysid], it->second);
    _systems[message.sysid]->system_impl()->set_system_id(message.sysid);
    _systems.erase(it);
}
```

But this part I removed is used only when no system 0 was found in `_systems` and `_is_single_system` is true (when `setup_udp_remote` was used).
Thus, it seems to me to be swapping the first system in `_systems `(which cannot be 0), by the system ID of the current message.

Reply:
Ok I see, this what is done for the first condition, when it founds a system 0:
```
if (it != _systems.end()) {
    std::swap(_systems[message.sysid], it->second);
    _systems[message.sysid]->system_impl()->set_system_id(message.sysid);
    _systems.erase(it);
}
```

But this part I removed is used only when no system 0 was found in `_systems` and `_is_single_system` is true (when `setup_udp_remote` was used).
Thus, it seems to me to be swapping the first system in `_systems `(which cannot be 0), by the system ID of the current message.

Reply:
It was added for the `setup_udp_remote` indeed, [here](https://github.com/mavlink/MAVSDK/pull/774/files#diff-1dc7f324308bcafacd5b550b936eedb0818ac54d2b094e89a8d184a3b1759a12R94-R100)